src/gsy_e/models/strategy/energy_parameters/energy_params_eb.py

```diff
@@ -77,31 +243,70 @@ def event_traded_energy(
         for time_slot, energy_value_kWh in trade_profile.items():
             self._state.decrement_energy_requirement(
                 purchased_energy_Wh=energy_value_kWh * 1000,
-                time_slot=time_slot,
+                time_slot=time_slot.set(tz=pendulum.UTC),
                 area_name=self._area.name)
 
 
-class ProductionStandardProfileEnergyParameters:
+class ProductionStandardProfileEnergyParameters(ForwardEnergyParams):
     """Energy parameters of the PV strategy for the forward markets."""
 
     def __init__(self, capacity_kW):
+        super().__init__()
         self.capacity_kW: float = capacity_kW
 
         self._state = PVState()
         self._area = None
         self._profile_generator: Optional[ForwardTradeProfileGenerator] = None
 
+    @property
+    def state(self) -> PVState:
+        """Retrieve the internal state of the asset."""
+        return self._state
+
     def serialize(self):
         """Return dict with the current energy parameter values."""
         return {
             "capacity_kW": self.capacity_kW
         }
 
+    @property
+    def capacity_kWh(self):
+        """Capacity of the PV in kWh."""
+        return convert_kW_to_kWh(self.capacity_kW, self._area.config.slot_length)
+
+    def get_available_energy_kWh(
+            self, market_slot: pendulum.DateTime, market_type: AvailableMarketTypes):
+        """Get the available offer energy of the PV."""
+        if market_type == AvailableMarketTypes.INTRADAY:
+            self._state.get_available_energy_kWh(market_slot)
+        elif market_type == AvailableMarketTypes.DAY_FORWARD:
+            return min(
+                self._state.get_energy_production_forecast_kWh(slot)
+                for slot in self.day_forward_slots(market_slot)
+            )
+
+        reference_slot = market_slot.set(hour=12, minute=0)
+
+        if self._state.get_energy_production_forecast_kWh(
+                reference_slot) <= FLOATING_POINT_TOLERANCE:
+            scaling_factor = 0.
+        else:
+            scaling_factor = (
+                    self._state.get_available_energy_kWh(reference_slot) /
+                    self._state.get_energy_production_forecast_kWh(reference_slot))
+        return abs(scaling_factor) * self.capacity_kWh
```

Comment:
I know, I think you explained this once already to me, but I still need to ask again.
AFAIU, the capacity of the PV is scaled with the available_energy.
So, if the 
production_forecast_kWh = 2
available_energy_kWh = 1 (1 was already traded)
the scaling factor is 0.5
What is the capacity_kWh here? It is the integral over production curve, right? So it should be grater that 2 kWh, right?
Thanks a lot for bearing with me!

Reply:
This algorithm is probably an overkill, however let me elaborate a bit further on the rationale behind this. First of all, all these are peak values and not integral values, including capacity_kWh. The reason for adding this part of the code, was because I needed to be able to calculate the scaled available peak energy for a duration of time (depending on the market type). In order to calculate this, I wanted to have one time slot as reference, which I knew that had non-zero production / consumption, and for all our profiles the most straightforward option is to use 12:00 as a reference point. However, it might be that the peak / max energy of the profile (essentially the capacity_kWh) is not at 12:00, but at some other time during the day. In order to get the scaled peak value (according to the already traded energy) I am calculating the scaling factor at 12:00 (essentially the ratio of available energy / production forecast at 12:00), and using this scaling factor, I calculate the peak of the available energy profile by multiplying with the peak value of the production profile. 

If any of the above does not make sense or is not clear let me know to further clarify this, thanks!

Reply:
Ok, so if the peak of the production curve is at noon, the following applies:

```python
self._state.get_energy_production_forecast_kWh("12:00") = self.capacity_kWh
```
?
I think I start to understand now. You want to get the available_energy_kWh for any market slot (e.g. 10:15) from the one at noon, because of the mentioned reasons. You do not care about the other market slots on the day, you only want to read and write to the reference_slot. Right?

Reply:
> Ok, so if the peak of the production curve is at noon, the following applies:

> self._state.get_energy_production_forecast_kWh("12:00") = self.capacity_kWh

Exactly! I think that we have reached a common understanding, however just to make it 100% clear: the reason for this calculation is that the peak of the profile (essentially the capacity_kWh) might not be at 12:00 but at 10:15 to use your example. Therefore I am calculating the "scaling factor" using an arbitrary timeslot which I know will have non-zero energy value (12:00 typically) and then multiply the self.capacity_kWh (which is really the profile peak).

As an aftermath of our discussion, I think that the `capacity_kWh` naming is a bit problematic here. If you agree I will rename it to `peak_energy_kWh` in order to be more understandable for the reader. Thanks!

Reply:
I think it would be useful for our future selves to add part of this conversation to the code in form of comments/docstrings!

Reply:
Will do so thanks!